docs-md/operate-and-deploy/ksql-vs-ksqldb.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: page
+title: KSQL versus ksqlDB
+tagline: KSQL versus ksqlDB
+description: Key differences between KSQL and ksqlDB.
+---
+
+!!! note
+    For the purposes of this topic, "ksqlDB" refers to ksqlDB 0.6.0 and beyond,
+    and "KSQL" refers to all previous releases of KSQL (5.3 and lower).
+
+ksqlDB is not backward compatible with previous versions of KSQL. This means
+that, ksqlDB doesn't run over an existing KSQL deployment. This topic provides
+an overview of each incompatibility, as well as guidance around migrating a
+KSQL deployment to ksqlDB.
+
+- [Syntax](#syntax)
+- [Storage](#storage)
+- [HTTP Response Format](#http-response-format)
+- [SHOW TOPICS](#show-topics)
+
+Syntax
+------
+
+With the introduction of a new query form, pull queries, ksqlDB's syntax had to
+evolve such that pull queries and push (streaming) queries can be expressed
+unambiguously. As a result, one of these two fundamental query forms required
+new syntax in order to be identified. Although push queries have existed since
+KSQL's inception, we ultimately decided to add new syntax to express them in
+ksqlDB so that pull query syntax would remain as standard as possible, because
+pull queries are much more likely to be used by external SQL-based systems that
+may be integrated with ksqlDB. For more information, see
+[KLIP-8: Queryable State Stores](https://github.com/confluentinc/ksql/blob/master/design-proposals/klip-8-queryable-state-stores.md).
+
+This syntax addition is incompatible with previous versions of KSQL. This means
+that **a streaming query in KSQL will not be interpreted correctly as a streaming query in ksqlDB**.
+Streaming queries in ksqlDB require the `EMIT CHANGES` modifier, which informs
+ksqlDB that the caller wants to receive incremental changes to the query result
+as it runs perpetually.
+
+The following statements show a comparison between a simple streaming query in
+KSQL versus an equivalent streaming query in ksqlDB:
+
+```sql
+-- KSQL:
+SELECT * FROM my_stream;
+
+-- ksqlDB:
+SELECT * FROM my_stream EMIT CHANGES;
+```
+
+While streaming KSQL queries will not be interpreted as streaming queries in
+ksqlDB, we have taken measures to mitigate the impact of this incompatibility
+to the greatest extent possible. In particular, **persistent streaming queries
+don't require the `EMIT CHANGES` modifier**. All persistent queries are
+interpreted implicitly as if they include an `EMIT CHANGES` modifier. This
+implicit behavior means that persistent queries created by using KSQL will run
+properly, as is, in ksqlDB.
+
+!!! note
+    `EMIT CHANGES` will be required for persistent queries in the future. We
+    recommend migrating all of your persistent streaming queries to include
+    the EMIT CHANGES modifier.
+
+Storage
+--------
+
+ksqlDB introduces changes to the underlying materialization storage that are
```

Comment:
I don't understand. What were the changes? I just spun up a 5.3 KSQL server, submitted some queries, then stood up ksqlDB from `master` and it came up fine. Are you saying that wasn't supposed to happen?

Reply:
In general it's not expected to work. The test you did probably just happened to be compatible with `0.6.0`. @big-andy-coates can provide more details.

Reply:
All of the compatibility breaking config defs have been disabled in master (and ksqlDB).  So we can't guarantee any backward compatibility with running queries from previous versions.

It's possible queries started on 5.3 / 5.4 currently work with 0.6, but that's by chance, not by design.